File: src/main/java/com/jiro4989/tkfm/model/RectangleModel.java
package com.jiro4989.tkfm.model;

import javafx.beans.property.DoubleProperty;
import javafx.beans.property.SimpleDoubleProperty;

public class RectangleModel {
  private final DoubleProperty width = new SimpleDoubleProperty(0.0);
  private final DoubleProperty height = new SimpleDoubleProperty(0.0);

  public RectangleModel(double w, double h) {
    setWidth(w);
    setHeight(h);
  }

  public DoubleProperty widthProperty() {
    return width;
  }

  public DoubleProperty heightProperty() {
    return height;
  }

  public double getWidth() {
    return width.get();
  }

  public double getHeight() {
    return height.get();
  }

  public void setWidth(double w) {
    this.width.set(w);
  }

  public void setHeight(double h) {
    this.height.set(h);
  }
}

File: src/main/kotlin/com/jiro4989/tkfm/model/RectangleModel.kt
package com.jiro4989.tkfm.model

import javafx.beans.property.DoubleProperty
import javafx.beans.property.SimpleDoubleProperty as DP

data class RectangleModel(val widthProperty: DoubleProperty, val heightProperty: DoubleProperty) {
  var width
    get() = widthProperty.get()
    set(value) = widthProperty.set(value)
  var height
    get() = heightProperty.get()
    set(value) = heightProperty.set(value)

  constructor(w: Double, h: Double) : this(DP(w), DP(h))

  operator fun div(scale: Double) = Pair(width / scale, height / scale)
}
